src/build.ts

```diff
@@ -54,36 +77,32 @@ const createFunctionFile = async (
   const fnFileRelative = path.join("functions", `${fnId}.js`);
   const fnBundledPath = path.join(outputDirectory, fnFileRelative);
 
-  // We'll default to just using whatever Deno executable is on the path
-  // Ideally we should be able to rely on Deno.execPath() so we make sure to bundle with the same version of Deno
-  // that called this script. This is perhaps a bit overly cautious, so we can look to remove the defaulting here in the future.
-  let denoExecutablePath = "deno";
   try {
-    denoExecutablePath = Deno.execPath();
-  } catch (e) {
-    protocol.error("Error calling Deno.execPath()", e);
-  }
-
-  try {
-    // call out to deno to handle bundling
-    const p = Deno.run({
-      cmd: [
-        denoExecutablePath,
-        "bundle",
-        "--quiet",
-        fnFilePath,
-        fnBundledPath,
-      ],
+    await DenoBundler.bundle({
+      entrypoint: fnFilePath,
+      outFile: fnBundledPath,
     });
+  } catch (denoBundleErr) {
+    if (!(denoBundleErr instanceof BundleError)) {
+      protocol.error(`Error bundling function file "${fnId}" with Deno`);
+      throw denoBundleErr;
+    }
+
+    protocol.warn(
+      "Failed bundling with `Deno Bundle` falling back to esbuild",
+    );
 
-    const status = await p.status();
-    p.close();
-    if (status.code !== 0 || !status.success) {
-      throw new Error(`Error bundling function file: ${fnId}`);
+    try {
+      const bundle = await EsbuildBundler.bundle({
+        entrypoint: fnFilePath,
+        absWorkingDir: workingDirectory,
+        configPath: await resolveDenoConfigPath(workingDirectory),
+      });
+      await Deno.writeFile(fnBundledPath, bundle);
```

Comment:
If we leave it in the `try` block it will more closely mimic the behavior of the `deno bundle` command since it wrote to disk and would fail if there were a file-writing issue